### PR TITLE
[ipoe.c] arp-reply not send with ip_unnumbering is enabled.

### PR DESCRIPTION
**Pull request by mort1k**
```
    100.64.0.1.67 > 11.116.22.130.68: BOOTP/DHCP, Reply, length 310, xid 0xfa31fff8, Flags [none]
	  Your-IP 11.116.22.130
	  Client-Ethernet-Address 50:0f:f5:a3:ed:f8
	  Vendor-rfc1048 Extensions
	    Magic Cookie 0x63825363
	    DHCP-Message (53), length 1: ACK
	    Server-ID (54), length 4: 100.64.0.1
	    Lease-Time (51), length 4: 600
	    RN (58), length 4: 300
	    RB (59), length 4: 525
	    Default-Gateway (3), length 4: 100.64.0.1
	    Subnet-Mask (1), length 4: 255.255.255.255
	    Domain-Name-Server (6), length 8: 8.8.8.8,8.8.4.4
	    Agent-Information (82), length 18: 
	      Circuit-ID SubOption 1, length 6: ^@^D^@i^@^E
	      Remote-ID SubOption 2, length 8: ^@^FM-HM->^YM-tUM-0
16:52:22.205319 50:0f:f5:a3:ed:f8 > ff:ff:ff:ff:ff:ff, ethertype 802.1Q (0x8100), length 64: vlan 105, p 0, ethertype ARP (0x0806), Ethernet (len 6), IPv4 (len 4), Request who-has 11.116.22.130 tell 91.216.22.130, length 46
16:52:22.210347 50:0f:f5:a3:ed:f8 > ff:ff:ff:ff:ff:ff, ethertype 802.1Q (0x8100), length 64: vlan 105, p 0, ethertype ARP (0x0806), Ethernet (len 6), IPv4 (len 4), Request who-has 11.116.22.130 tell 0.0.0.0, length 46
16:52:22.216972 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 114: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 56) :: > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 2 group record(s) [gaddr ff02::1:ff2f:a115 to_ex, 0 source(s)] [gaddr ff02::2 to_ex, 0 source(s)]
16:52:22.736967 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 90: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 255, next-header ICMPv6 (58) payload length: 32) :: > ff02::1:ff2f:a115: [icmp6 sum ok] ICMP6, neighbor solicitation, length 32, who has fe80::2e0:edff:fe2f:a115
	  unknown option (14), length 8 (1): 
	  0x0000:  dbb0 ebc3 1609
16:52:23.024965 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 114: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 56) :: > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 2 group record(s) [gaddr ff02::1:ff2f:a115 to_ex, 0 source(s)] [gaddr ff02::2 to_ex, 0 source(s)]
16:52:23.760972 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 134: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 76) fe80::2e0:edff:fe2f:a115 > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 3 group record(s) [gaddr ff02::1:ff00:0 to_ex, 0 source(s)] [gaddr ff02::1:ff2f:a115 to_ex, 0 source(s)] [gaddr ff02::2 to_ex, 0 source(s)]
16:52:23.772965 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 94: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 36) fe80::2e0:edff:fe2f:a115 > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 1 group record(s) [gaddr ff02::1:ff00:0 to_ex, 0 source(s)]
16:52:24.080964 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 94: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 36) fe80::2e0:edff:fe2f:a115 > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 1 group record(s) [gaddr ff02::1:ff00:0 to_ex, 0 source(s)]
16:52:24.236964 00:e0:ed:2f:a1:15 > 50:0f:f5:a3:ed:f8, ethertype 802.1Q (0x8100), length 134: vlan 105, p 0, ethertype IPv6 (0x86dd), (hlim 1, next-header Options (0) payload length: 76) fe80::2e0:edff:fe2f:a115 > ff02::16: HBH (rtalert: 0x0000) (padn) [icmp6 sum ok] ICMP6, multicast listener report v2, 3 group record(s) [gaddr ff02::1:ff00:0 to_ex, 0 source(s)] [gaddr ff02::1:ff2f:a115 to_ex, 0 source(s)] [gaddr ff02::2 to_ex, 0 source(s)]
```
And yes, Some routers send arp after configuration interface as unnumbering. In this situation accel  not response to client on arp  request.

### CODE REVIEW

**Comment by DmitriyEshenko** [comment]
Please describe why you need this exactly?

**Comment by mort1k** [comment]
I think unnumbered option not use arp protocol for host resolve.

**Comment by DmitriyEshenko** [comment]
ip-unnumbered just create /32 routes on server side as I know, but client still get properly mask like /24 and etc.
The main reason why I\m asking, because ip-unnumbered by default is active.

**Comment by mort1k** [comment]
What you think about interpretation unnumbering is: 
   both side use netmask /32 and not use arp and correlation with option shared.

**Comment by svlobanov** [comment]
how client will known gw mac address without arp?

**Comment by mort1k** [comment]
one use for this is /32 DHCP routing. The specific example is that you have a /24 or larger block, but you wish to have all of your customers come back to your router for routing, no ARPs no layer 2 connectivity to any device on the network.

Its really simple to enable this, by changing your DHCP server netmask to 32
However, you also need a DHCP option 121, to point the customer to the classless route, in our case the default gateway. This spreadsheet will allow you to do this (the one in this thread) by simply calculating the value of the DHCP option 121. Then you need to put that option on the network that you wish to use. Note that that option value will be DIFFERANT for each subnet you have.

So simply put to do /32 DHCP routing, with a prefix larger than /32, so lets say a /21 or even a /16. You can change the netmask to 32 in the dhcp networks tab, then add the option 121 to that network.

**Comment by mort1k** [comment]
So,
  if unnumbered option is enabled, option 121 should to be send in DHCP response with  client  netmask /32.

https://www.medo64.com/2018/01/configuring-classless-static-route-option/

**Comment by mort1k** [comment]
updated first message with traffic dump accel not response on arp request. 
Debain12, accel-ppp from git master, proxy-arp=0, unnumbered-ip=0, gw-ip-address = ip/32

**Comment by mort1k** [comment]
Decode attribute is  not correct  <Classless-Route 0.0.0.0/0 via 64.0.1.15>

**Comment by DmitriyEshenko** [comment]
It is pretty strange usage, and I think it does not related to ip-unnumbered, because it is Classless Routing. Pay attention that Win devices dos not know 121 and use Option 249 instead. So, from my side I think that this PR is not ready to merge, because it will destroy basic functionality

**Comment by mort1k** [comment]
Same things client must get /32 mask if option unnumbering is enabled.

https://github.com/accel-ppp/accel-ppp/pull/104

**Comment by mort1k** [comment]
https://www.juniper.net/documentation/us/en/software/junos/subscriber-mgmt-vlan/topics/ref/statement/host-prefix-only-edit-interfaces-unit.htm
